components/workspace-rollout-job/pkg/analysis/analyzer.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package analysis
+
+import (
+	"context"
+)
+
+type Decision int
+
+const (
+	DecisionRevert      Decision = -1
+	DecisionNoData      Decision = 0
+	DecisionMoveForward Decision = 1
+)
+
+type Analyzer interface {
+	// Given a cluster name, MoveForward is called by the rollout routine
+	// repeatedly to determine whether to move forward on the rollout or not.
+	// DecisionRevert means to revert
+	// DecisionNoData means that there is no data to make an informed decision
+	// DecisionMoveForward means to move forward based on data
```

Comment:
We can probably remove those comments now that we're not using -1, 0 and 1. The types are self explanatory 😛